lib/presentation/app.dart

```diff
@@ -41,6 +44,19 @@ class VostApp extends StatelessWidget {
             ),
         routeIntro: (context) => IntroPage(bloc: DependencyProvider.of(context).getIntroBloc()),
       },
+      onGenerateRoute: (routeSettings) {
+        switch(routeSettings.name) {
+          case routeDetails:
+            var occurrence = routeSettings.arguments as OccurrenceModel;
```

Comment:
Having `OcurrentModel occurrent = routeSettings.arguments` is more clean IMO or at least having the `var` replaced for `OccurrenceModel `.